### ISSUE TITLE
Link modification and triggers

### ISSUE THREAD

**Issue by samseaver** [ModelSEED/ModelSEED#119]
I know how to update properties of an object within the biochemistry, but this is the first time I've tried updating the uuid:

```
my $rxn=$biochem->getObjectByAlias('reactions','R00001','KEGG');
foreach my $rgt (@{$rxn->reagents()}){
    if($rgt->compound_uuid() eq $old_cpd->uuid()){
        $rgt->compound_uuid($new_cpd_uuid());
    }
}
```

Would this change in uuid cause problems at a lower-level?

**Comment by devoid** [comment]
That should work as expected. What **doesn't** work is doing:

``` perl
if ($rgt->compound_uuid eq $old_cpd->uuid) {
    $rgt->compound($new_cpd);
}
```

Because this data gets lost during the save process. I'm thinking we can fix this with a trigger on the update of the `compound` attribute. An additional discrepancy: in your case, accessing `$rgt->compound()` inside that if condition may not return the new compound. This should be fixed by adding an additional trigger when the uuid is changed.

I'll look into implementing this.

TL;DR What you have will work, but don't expect the _link_ `compound()` to reflect `compound_uuid()` or visa versa.

**Comment by samseaver** [comment]
In fact, I was exploring what I believed was a bug until I saw this.  I may not expect `$rgt->compound()` to work, but I need it to.  As I change the compounds in a reaction, I need to compare the same reaction via its equation, but since `Reaction::createEquation()` uses the link `compound()`, I get an identical equation as before, which needs to be fixed.  I need to be able to discard reactions based on the updated equation so this isn't something that can wait until "after" saving the biochemistry.

How hard is this to fix?
